codec_test.go

```diff
@@ -1,20 +1,26 @@
-package jsonrpc2
+package jsonrpc2_test
```

Comment:
I converted this whitebox test to a blackbox test because that allowed me to reuse the `noopHandler` defined in `jsonrpc2_test.go`